File: MidTermVarshaHunjanApp/app/src/main/java/com/example/midtermvarshahunjanapp/MainActivity.java
package com.example.midtermvarshahunjanapp;
import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    private EditText literInput;
    private Button convertButton;
    private TextView resultText;

    private void convertLitersToCups() {
        try {
            double liters = Double.parseDouble(literInput.getText().toString());
            double cups = liters * 4.22;
            resultText.setText(String.format("%.2f liters is equal to %.2f cups", liters, cups));
        } catch (NumberFormatException e) {
            resultText.setText("Please enter a valid number of liters.");
        }

    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);


        literInput = findViewById(R.id.literInput);
        convertButton = findViewById(R.id.convertButton);
        resultText = findViewById(R.id.resultText);

        convertButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                convertLitersToCups();
            }
        });

    }

}

File: HelloWorldKotlinApp/app/src/main/java/com/example/helloworldkotlinapp/MainActivity.kt
package com.example.helloworldkotlinapp

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.animation.Animation
import android.view.animation.AnimationUtils
import android.widget.Button
import android.widget.TextView

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)


        val buttonZoomIn: Button = findViewById(R.id.click_me_button)
        val textView: TextView = findViewById(R.id.hello)

        buttonZoomIn.setOnClickListener() {

            // loading the animation of
            // zoom_in.xml file into a variable
            val animZoomIn = AnimationUtils.loadAnimation(this,
                R.anim.zoom_in)

            animZoomIn.duration = 5000


                textView.startAnimation(animZoomIn)
        }
    }

}